cmd/api/http/handlers.go

```diff
@@ -122,6 +122,14 @@ func (h *BookHandler) createBook(w http.ResponseWriter, r *http.Request) {
 
 	storedBook, err := h.bookService.CreateBook(r.Context(), reqBook)
 	if err != nil {
+		if errors.Is(err, r.Context().Err()) {
```

Comment:
This way we can catch a timed out context as well a canceled one.

Reply:
You can have two types of Context error => https://pkg.go.dev/context#pkg-constants.
When context is cancelled the request was `Aborted`.
When context deadline was exceeded its a `Timeout`.

`context.Cancelled`
`context.DeadlineExceeded`

Reply:
I put both error types at `http_test.go` at the new commit to we see how it works.